resolver.go

```diff
@@ -145,6 +145,14 @@ func (r *resolver) Start() error {
 		return fmt.Errorf("setting up IP table rules failed: %v", err)
 	}
 
+	// If the container goes down quickly after start resolver.Stop() can
+	// get called before Start() completes. Check if the socket is still
+	// valid before Activating the DNS server. This avoids a panic in dns
+	// libarary code because of missing nil check for interface's data.
+	// github.com/docker/docker/issues/28112
+	if r.conn == nil {
```

Comment:
`wouldn't the same issue be there also when we construct the tcpServer ?` - with this change we will return from `Start()` if resolver's UDPConn is nil. Its not possible to have `resolver.conn != nil &&  resolver.tcpListen == nil`. But I think there is no harm in checking for resolver.tcpListen as well. Will change it.